src/requests/cancel-tx.go

```diff
@@ -0,0 +1,54 @@
+package requests
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"github.com/ethereum/go-ethereum/crypto"
+	"log"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+func SendCancelTx(client *ethclient.Client, privateKey *ecdsa.PrivateKey) {
```

Comment:
This function will create a transaction from the encrypting RPC's private key. So it will not cancel the user's transaction. Instead the user transaction has to be forwarded to the actual RPC behind.

Reply:
Updated to use a function that sends the user's transaction directly [here](https://github.com/shutter-network/encrypting-rpc-server/pull/14/files#diff-a149120ffb5074c9697008d488ab4ed0280816c09365253edb2d20e9ae7bf9ecR10).